src/visitor.rs

```diff
@@ -402,3 +371,42 @@ impl<'a> FmtVisitor<'a> {
         }
     }
 }
+
+impl<'a> Rewrite for [ast::Attribute] {
+    fn rewrite(&self, context: &RewriteContext, _: usize, offset: usize) -> Option<String> {
+        let mut result = String::new();
+        if self.len() == 0 {
+            return Some(result);
+        }
+        let indent = utils::make_indent(offset);
+
+        for (i, a) in self.iter().enumerate() {
+            let a_str = context.snippet(a.span).unwrap();
+
+            if i > 0 {
+                let comment = context.snippet(codemap::mk_sp(self[i-1].span.hi, a.span.lo)).unwrap();
+                // This particular horror show is to preserve line breaks in between doc
+                // comments. An alternative would be to force such line breaks to start
+                // with the usual doc comment token.
+                let multi_line = a_str.starts_with("//") && comment.matches('\n').count() > 1;
+                let comment = comment.trim();
+                if comment.len() > 0 {
+                    result.push_str(&indent);
+                    result.push_str(comment);
```

Comment:
I think not. If memory serves, a doc comment is just sugar for an attribute of the form `#[comment("Comment body")]`.

This code deals with normal comments in between the attributes.